devito/ops/node_factory.py

```diff
@@ -0,0 +1,214 @@
+
+        Parameters
+        ----------
+        name : str
+            Name of the symbol to be created.
+
+        Returns
+        -------
+        :class:`Symbol` 
+            Object :class:`Symbol` with the given name.
+        """
+        return Symbol(name=name)
+
+    def new_int_node(self, number):
+        """
+        Creates a new sympy :class:`Integer` object.
+
+        Parameters
+        ----------
+        number : int
+            Number to initialize an Integer node.
+
+        Returns
+        -------
+        :class:`Integer`
+            Object of :class:`Integer`.
+        """
+        return Integer(number)
+
+    def new_float_node(self, number):
+        """
+        Creates a new sympy :class:`float` object.
+
+        Parameters
+        ----------
+        number : float
+            Number to initialize an Float node.
+
+        Returns
+        -------
+        :class:`Float`
+            Object of :class:`Float`.
+        """
+        return Float(number)
+
+    def new_rational_node(self, num, den):
+        """
+        Creates a new sympy :class:`Rational` object to represent 
+        a mathematical rational.
+
+        Parameters
+        ----------
+        num : float
+            Rational numerator.
+        den : float
+            Rational denominator.
+
+        Returns
+        -------
+        :class:`Rational`
+            Object :class:`Rational`. 
+        """
+        return Rational(num, den)
+
+    def new_add_node(self, lhs, rhs):
+        """
+        Represents a new mathematical sum node operation.
+
+        Parameters
+        ----------
+        lhs 
+            The left hand side of the sum operation.
+        rhs : [type]
+            The right hand side of the sum operation.
+
+        Returns
+        -------
+        :class:`Add`
+            Object of :class:`Add`.
+        """
+        return Add(lhs, rhs)
+
+    def new_mul_node(self, lhs, rhs):
+        """
+        Creates a new mathematical multiplication node operation.
+
+        Parameters
+        ----------
+        lhs
+            The left hand side of the multiplication.
+        rhs : [type]
+            The right hand side of the multiplication.
+
+        Returns
+        -------
+        :class:`Mul`
+            Object of :class:`Mul`.
+        """
+        return Mul(lhs, rhs)
+
+    def new_divide_node(self, num, den):
+        """
+        Creates a new division node operation.
+
+        Notes
+        -----
+        The division is represented by the multiplicating the numerator with 
+        denominator's inverse.
+
+        Parameters
+        ----------
+        num
+            The division numerator.
+        den : [type]
+            The division denominator.
+
+        Returns
+        -------
+        :class:`Mul`
+            Object of :class:`Mul` representing the division.
+        """
+        return Mul(num, Pow(den, Integer(-1)))
+
+    def new_grid(self, devito_indexed, dimensions):
```

Comment:
do not call it "grid". "grid" was YASK terminology, no need to use the same here. ``new_indexed`` is probably fine.

and instead of ``devito_indexed``, just ``indexed``

Reply:
new_indexed i want to use in the end, when i create the new Indexed object.
For the Array obj i think grid is a good name, because it does represent a grid.
Will change the devito_indexed to indexed.